google/cloud/security/notifier/pipelines/email_violations_pipeline.py

```diff
@@ -70,34 +71,24 @@ def _get_output_filename(self):
                                                      output_timestamp)
         return output_filename
 
-    def _write_temp_attachment(self):
-        """Write the attachment to a temp file.
-
-        Returns:
-            str: The output filename for the violations json just written.
-        """
-        # Make attachment
-        output_file_name = self._get_output_filename()
-        output_file_path = '{}/{}'.format(TEMP_DIR, output_file_name)
-        with open(output_file_path, 'w+') as f:
-            f.write(parser.json_stringify(self.violations))
-        return output_file_name
-
     def _make_attachment(self):
         """Create the attachment object.
 
         Returns:
             attachment: SendGrid attachment object.
         """
-        output_file_name = self._write_temp_attachment()
+        tmp_file = tempfile.NamedTemporaryFile(delete=False)
```

Comment:
This would be better using the "with" clause.

Reply:
Also, what if this raises an exception, e.g. disk is full. Can we catch this properly and handle it? Otherwise the references below to tmp_file.name won't work.

Reply:
I can change the code to use a `with` clause. Do I need to open an issue for that or shall I just prepare a quick PR without it?

Reply:
Since I merged the PR before seeing Michael's comment, please just submit a follow-up PR.  The code should look something like:

```
        with tempfile.NamedTemporaryFile(delete=False) as tmp_file:
            tmp_file.write(parser.json_stringify(self.violations))

        attachment = self.mail_util.create_attachment(
            file_location=tmp_file.name,
            content_type='text/json',
            filename=self._get_output_filename(),
            disposition='attachment',
            content_id='Violations'
        )
        os.unlink(tmp_file.name)
        return attachment
```